cmd/queue/main.go

```diff
@@ -232,13 +233,16 @@ func main() {
 		logger.Fatalw("Failed to create stats reporter", zap.Error(err))
 	}
 
+	protoStatReporter := queue.NewProtobufStatsReporter(env.ServingPod, reportingPeriod)
+
 	reportTicker := time.NewTicker(reportingPeriod)
 	defer reportTicker.Stop()
 
 	stats := network.NewRequestStats(time.Now())
 	go func() {
 		for now := range reportTicker.C {
 			promStatReporter.Report(stats.Report(now))
+			protoStatReporter.Report(stats.Report(now))
```

Comment:
`stats.Report` resets the stat reporting. You'll want:

```golang
stat := stats.Report(now)
promStatReporter.Report(stat)
protoStatReporter.Report(stat)
```

I'm fairly sure that's the bug that causes the test failure.

Reply:
Yeah I was thinking of that part because if it does not get the right stats it will not scale. Let me try it.